### PR TITLE
Update maps server dockerfile to two stages

### PR DESCRIPTION
**Pull request by junqiu-lei**
Signed-off-by: Junqiu Lei <junqiu@amazon.com>

### Description
Update maps server dockerfile to two stages and update download tar file name.
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/2681
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch-maps-server/entrypoint.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ if [ "${1}" = "import" ]; then
     cd public/tiles/data
     if [ -n "${DOWNLOAD_TILES:-}" ]; then
         echo "INFO: Download Tiles images: $DOWNLOAD_TILES"
-        wget "$DOWNLOAD_TILES" -O tiles.tar.gz
+        curl -SL -o tiles.tar.gz "$DOWNLOAD_TILES"
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply opensearch-maps-server/entrypoint.sh:24]: You can ignore this if you have other things planned around this, just curious.

File: opensearch-maps-server/Dockerfile [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
-    yum install -y nodejs wget tar gzip unzip python37 && \
-    yum clean all
+COPY --from=stage_0 /src/GeoJSONData public/vectors/data
+COPY --from=stage_0 /src/oss_compliance .
 
-RUN pip3 install --no-cache --upgrade \
-    "requests>=2.18.4,<=2.22.0"
+RUN curl -SL https://rpm.nodesource.com/setup_16.x | bash - && \
+    yum install -y nodejs-16.17.1  python3 && yum clean all
+RUN pip3 install requests
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply opensearch-maps-server/Dockerfile:33]: Thanks, will update to use a specific version.

File: opensearch-maps-server/entrypoint.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply opensearch-maps-server/entrypoint.sh:24]: I prefer to keep it as it is. It's the only place to use it, and user can optionally not provide env `DOWNLOAD_TILES ` instead to use their own tiles set by `-v /absolute/path/to/tiles/:/usr/src/app/dist/public/tiles/data/`

File: opensearch-maps-server/Dockerfile [hunk 3]
```diff
@@ -1,33 +1,48 @@
...
-RUN curl -sL https://rpm.nodesource.com/setup_16.x | bash - && \
-    yum install -y nodejs wget tar gzip unzip python37 && \
-    yum clean all
+COPY --from=stage_0 /src/GeoJSONData public/vectors/data
+COPY --from=stage_0 /src/oss_compliance .
 
-RUN pip3 install --no-cache --upgrade \
-    "requests>=2.18.4,<=2.22.0"
+RUN curl -SL https://rpm.nodesource.com/setup_16.x | bash - && \
+    yum install -y nodejs-16.17.1  python3 && yum clean all
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply opensearch-maps-server/Dockerfile:32]: Thanks, will update.

**Review by peterzhuamazon — APPROVED** [review]

**Review by VijayanB — APPROVED** [review]

**Pull request merged**